test/types.ts

```diff
@@ -4,148 +4,86 @@
  */
 
 import {
-  expectTypeOf,
-} from 'expect-type';
+  expectType,
+} from 'tsd';
+import {
+  z,
+} from 'zod';
 import {
   createPool,
   sql,
 } from '../src';
 import {
-  type PrimitiveValueExpression,
   type QueryResult,
 } from '../src/types';
 
+const Row = z.object({
+  foo: z.string(),
+});
+
 export const queryMethods = async (): Promise<void> => {
   const client = createPool('');
 
-  type Row = {
-    bar: boolean,
-    foo: string,
-  };
-
   // any
   const any = await client.any(sql``);
-  expectTypeOf(any).toEqualTypeOf<ReadonlyArray<Record<string, PrimitiveValueExpression>>>();
+  expectType<readonly any[]>(any);
 
-  const anyTyped = await client.any<Row>(sql``);
-  expectTypeOf(anyTyped).toEqualTypeOf<readonly Row[]>();
-
-  const anyTypedQuery = await client.any(sql<Row>``);
-  expectTypeOf(anyTypedQuery).toEqualTypeOf<readonly Row[]>();
+  const anyTyped = await client.any(sql.type(Row)``);
```

Comment:
I don't think the old test should be deleted? Is slonik going to stop supporting `` sql<Row>`select 123` ``?? I would have thought this is just a new feature and should have corresponding new type tests. Right now this PR deletes many many tests that ensure slonik's types are correct for end users (since test/dts.ts also got deleted entirely).

Reply:
I don't see a path for supporting both. They don't mix will.

What happens if someone does:

```ts
sql<{id: number}>.type(
    z.object({
      id: z.string()
    })
)`select 123`
```

Reply:
Not a risk - that isn't valid ts syntax. The generic on `sql` is only when used as a function directly. `sql.type` is another, independent, generic function which returns a non-generic function. I'm very sure of this, and still pretty concerned about abandoning the existing (and totally satisfactory!) form slonik has now. If you're not convinced, can I have a few days to come up with a pull request which proves this out?

Reply:
> If you're not convinced, can I have a few days to come up with a pull request which proves this out?

Absolutely!